src/decisionengine_modules/NERSC/sources/NerscSFApi.py

```diff
@@ -62,43 +61,51 @@ def check_accesstoken(self, nersc_user):
         except KeyError:
             self.logger.error(f"Unknown user '{nersc_user}', exiting")
             return None
-
+        print(nersc_user)
         rawfile = params_dict["rawfile"]
         pemfile = params_dict["private_key"]
         clientidfile = params_dict["client_id_file"]
         with open(clientidfile) as cifile:
             client_id = cifile.read()
             client_id = client_id.rstrip()
-
+        atoken = None
         if not os.path.exists(rawfile):
             self.logger.debug(f"{rawfile} does not exist. Need to generate")
         else:
             atoken = None
             with open(rawfile) as afile:
                 atoken = afile.read()
                 atoken = atoken.rstrip()
-            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSignatureError
-            try:
-                jwt.decode(atoken, options={"verify_signature": False})
-                self.logger.debug("Access Token not expired. Returning without generating a new access token")
-                return atoken  # This means the existing access token is not expired.
-
-            except jwt.ExpiredSignatureError:
-                self.logger.debug("Access Token expired")
-
-        certs = pem.parse_file(pemfile)
-        private_key = str(certs[0])
-        client = OAuth2Session(
-            client_id=client_id, client_secret=private_key, token_endpoint_auth_method="private_key_jwt"
-        )
-        client.register_client_auth_method(PrivateKeyJWT(token_url))
-        resp = client.fetch_token(token_url, grant_type="client_credentials")
+            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSign
 
-        newtoken = resp["access_token"]
+        if atoken is not None:
+            rvalue = jwt.decode(atoken, options={"verify_signature": False})
```

Comment:
The intent of the PR is to work under EL9 /  JWT 2.4.  We do not intend to back-port this patch to de 1.7.0 / JWT 1.6 which is working fine as is.

Reply:
If that's the case, and if what I've suggested above works, then the below comment is not applicable, because you won't need to calculate the diff at all.

Reply:
Let's resolve this conversation for now.